core/src/main/java/bisq/core/grpc/CoreWalletsService.java

```diff
@@ -50,6 +75,68 @@ public long getAvailableBalance() {
         return balance.getValue();
     }
 
+    public long getAddressBalance(String addressString) {
+        Address address = getAddressEntry(addressString).getAddress();
+        return btcWalletService.getBalanceForAddress(address).value;
+    }
+
+    public String getFundingAddresses() {
+        if (!walletsManager.areWalletsAvailable())
+            throw new IllegalStateException("wallet is not yet available");
+
+        if (walletsManager.areWalletsEncrypted() && tempAesKey == null)
+            throw new IllegalStateException("wallet is locked");
```

Comment:
Checks of this type will probably be performed for most API endpoints. It would be nice to eventually see them moved into their own methods (`checkThatWalletsAreAvailable()`, etc.). Not requesting this change at this point.

Reply:
How about `isWalletUnlocked()` , returning `true` if the wallet is encrypted, and temporarily unlocked after a call to `unlockWallet(String password, long timeout)` ?

"Available" is taken;  it means the daemon is sufficiently warmed up and the btc & bsq wallets are not `null`. See `walletsManager.areWalletsAvailable()`.

Good call.  I'll create a new PR for this today.

Reply:
What I see as a reusable and complex unit is not the predicate, but the pair of a predicate and the throwing of the exception. For example, checking if a wallet is locked and, if so, throwing an exception saying that it is locked. So the method I'm suggesting  would look like this:
```
private void aGoodName() {
  if (walletsManager.areWalletsEncrypted() && tempAesKey == null
    throw new IllegalStateException("wallet is locked");
}
```

Does that make sense?

Reply:
I'm preparing to test this now:

```
   // Throws a RuntimeException if wallets are encrypted and locked.
    private void verifyEncryptedWalletIsUnlocked() {
        if (walletsManager.areWalletsEncrypted() && tempAesKey == null)
            throw new IllegalStateException("wallet is locked");
    }

```

Reply:
Requested change is in PR [4312](https://github.com/bisq-network/bisq/pull/4312).